### ISSUE TITLE
[Accessibility]: "Infer index configuration from HEAD" button should announce state change success/failure

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#44560]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33517

### Problem description

When a screen reader user clicks the "Infer index configuration from HEAD" button on the auto-index configuration page, there is no announcement whether the text was actually changed or not, i.e. whether inference failed or not. This requires scrolling back to the buffer. However, comparing the output (to figure out if the button succeeded or not) may be practically very time consuming if both the original and new config are more than a few words.

### Expected behavior

There should be a screen reader announcement whether the button succeeded or not.

### Additional details

https://user-images.githubusercontent.com/93103176/202496162-669847fa-00b2-4842-9283-ecabee61a3ca.mov

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by felixfbecker** [comment]
I think this is a duplicate of https://github.com/sourcegraph/sourcegraph/issues/35196
